### DISCUSSION TITLE
Can't access the Rest API docs

### DISCUSSION THREAD

**Discussion by selvinkuik** [RaspAP/raspap-webgui#1665] [Q&A]
I've just enabled the Rest API as per the instructions here: https://docs.raspap.com/restapi/

I can confirm it is working and I can query directly from the command line on my Raspberry Pi, e.g.

```
curl -X 'GET' \
  'http://raspberrypi.local:8081/system' \
  -H 'accept: application/json' \
  -H 'access_token: o2eycsnwzacgcukkdkxulmvcva7hou5q'
```

But when I click the link to the interactive Rest API docs, the link http://10.3.141.1:8081/docs doesn't work. I get ERR_CONNECTION_REFUSED. Am I missing a configuration step to allow access to port 8081? Or are the docs available to be viewed online somewhere?

I'm on RaspAP v3.1.7 with Raspberry Pi OS Lite (64-bit) on a Pi 5.

**Comment by billz** [comment]
I wasn't able to reproduce this on a clean 3.1.7 install. Since you're able to query the API via the console, the service is functioning and the port isn't in use by another process. Browser plugins that force HTTPS (such as HTTPS Everywhere) can result in the error you're seeing. Start by eliminating the obvious and try a different browser or disable plugins when accessing the API docs.

You can also verify that the uvicorn process is running:

```
$ ps aux | grep uvicorn
pi       2244  0.7  5.2  60688 48320   00:42   0:07 /usr/bin/python3 -m uvicorn main:app --host 0.0.0.0 --port 8081
```

RaspAP implements FastAPI, so if these steps don't resolve the issue your best bet is to search [this repo](https://github.com/fastapi/fastapi/) and the [uvicorn docs](https://www.uvicorn.org/).
